internal/action/setup.go

```diff
@@ -0,0 +1,287 @@
+	}
+
+	email, err = termio.AskForString(ctx, "📧 What is your email?", email)
+	if err != nil {
+		return err
+	}
+
+	passphrase := xkcdgen.Random()
+	pwGenerated := true
+	if bv, err := termio.AskForBool(ctx, "⚠ Do you want to enter a passphrase? (otherwise we generate one for you)", false); err != nil && bv {
+		pwGenerated = false
+		sv, err := termio.AskForPassword(ctx, "✍ Please enter your passphrase")
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read passphrase")
+		}
+		passphrase = sv
+	}
+
+	out.Yellow(ctx, "⏳ This can take a long time. If you get impatient see https://github.com/gopasspw/gopass/blob/master/docs/entropy.md")
+	if want, err := termio.AskForBool(ctx, "Continue?", true); err != nil || !want {
+		return errors.Wrapf(err, "User aborted")
+	}
+
+	if err := crypto.GenerateIdentity(ctx, name, email, passphrase); err != nil {
+		return errors.Wrapf(err, "failed to create new private key in batch mode")
+	}
+
+	out.Print(ctx, "✅ Key pair generated")
+
+	if pwGenerated {
+		out.Print(ctx, color.MagentaString("Passphrase: ")+color.HiGreenString(passphrase))
+	}
+
+	// avoid the gpg cache or we won't find the newly created key
+	kl, err := crypto.ListIdentities(gpg.WithUseCache(ctx, false))
+	if err != nil {
+		return errors.Wrapf(err, "failed to list private keys")
+	}
+	if len(kl) > 1 {
+		out.Print(ctx, "⚠ More than one private key detected. Make sure to use the correct one!")
+		return nil
+	}
+	if len(kl) < 1 {
+		debug.Log("Private Keys: %+v", kl)
+		return errors.New("failed to create a usable key pair")
+	}
+
+	// we can export the generated key to the current directory for convenience
+	if err := s.initExportPublicKey(ctx, crypto, kl[0]); err != nil {
+		return err
+	}
+	out.Green(ctx, "✅ Key pair validated")
+	return nil
+}
+
+func (s *Action) initExportPublicKey(ctx context.Context, crypto backend.Crypto, key string) error {
+	fn := key + ".pub.key"
+	want, err := termio.AskForBool(ctx, fmt.Sprintf("Do you want to export your public key to %q?", fn), false)
+	if err != nil {
+		return err
+	}
+	if !want {
+		return nil
+	}
+	pk, err := crypto.ExportPublicKey(ctx, key)
+	if err != nil {
+		return errors.Wrapf(err, "failed to export public key")
+	}
+	if err := ioutil.WriteFile(fn, pk, 06444); err != nil {
+		out.Error(ctx, "❌ Failed to export public key %q: %q", fn, err)
+		return err
+	}
+	out.Print(ctx, "✴ Public key exported to '%s'", fn)
+	return nil
+}
+
+func (s *Action) initHasUseablePrivateKeys(ctx context.Context, crypto backend.Crypto) bool {
+	kl, err := crypto.ListIdentities(gpg.WithAlwaysTrust(ctx, false))
+	if err != nil {
+		return false
+	}
+	return len(kl) > 0
+}
+
+func (s *Action) initSetupGitRemote(ctx context.Context, team, remote string) error {
+	var err error
+	remote, err = termio.AskForString(ctx, "Please enter the git remote for your shared store", remote)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read user input")
+	}
+	{
+		ctx := ctxutil.WithHidden(ctx, true)
```

Comment:
Why `WithHidden`?

Reply:
IIRC we didn't want possible clutter from the RCS ops.